alica_engine/include/engine/PlanPool.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <map>
+#include <memory>
+
+namespace alica
+{
+class AlicaEngine;
+class Plan;
+class BasicPlan;
+class ConfAbstractPlanWrapper;
+class IPlanCreator;
+class RunningPlan;
+class Configuration;
+
+class PlanPool {
```

Comment:
Why do you need this?
It looks to me `BasicPlans` can be tugged onto ModelElements just like conditions and utility functions. Why create yet another managing facility?